[REP-2004] add to the motivation to address common questions (#269)
ros-infrastructure/rep: rep-2004.rst

This REP describes a set of categories meant to convey the quality or maturity of packages in the ROS ecosystem.
Inclusion in a category, or quality level, is based on the policies to which a package adheres.
The categories address policies about versioning, change control, documentation, testing, dependencies, platform support and security.

Motivation
==========

The purpose of this REP is to provide standard guidelines for evaluating package quality, for both maintainers and consumers.

Package maintainers can use these guidelines to establish and document their own specific policies addressing how their packages achieve certain quality levels.

Consumers can use the guidelines in the REP, as well as the corresponding policies defined by maintainers, to set expectations on package quality and better evaluate whether a package or its dependencies meet the standards of use in their projects.

> By setting these goals and expectations for maintainers and consumers, respectively, the guidelines will also encourage better quality across the ROS ecosystem.

Reviewer: I think the last two sentences from the antigoal section belong here (feel free to ignore this one if you think they're better off below, I just think since they go back to talking about goals they shouldn't be in a section with an opposite purpose)

```suggestion
The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications. 

In turn, the documentation of policies and communication of quality characteristics between maintainers and consumers will improve.
By setting these goals and expectations for maintainers and consumers, respectively, the guidelines will also encourage better quality across the ROS ecosystem.
```

Antigoals
^^^^^^^^^

The motivation behind this REP does not include:

* Enforcing quality levels on maintainers

>     * No maintainer is required to abide by any of the guidelines in this REP

Reviewer: ```suggestion
  * No maintainer is required to abide by any of the guidelines in this REP
```

* Dictating policy implementation specifics necessary to achieve a quality level

    * Maintainers can come up with their own policies or follow an existing set of guidelines for achieving these quality levels, see `Reference Implementation`_ for examples of existing guidelines

>     * Policy requirements (which tools to use, specific thresholds, etc.) are *intentionally generic*

Reviewer: ```suggestion
  * Policy requirements (which tools to use, specific thresholds, etc.) are *intentionally generic*
```

* Enforcing the community to approve of policies documented by package maintainers

>     * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
>     * Enforcement cannot, therefore, be done in an automated fashion in all cases

Reviewer: ```suggestion
  * Policies that are skipped or altered with justifications require evaluation by the consumer to see if that reasoning is acceptable for their standards of use
  * Enforcement cannot, therefore, be done in an automated fashion in all cases
```

>     * Instead, peer reviewed lists of packages, where maintainers can take their `quality declaration <Reference Implementation>`_ and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 

Reviewer: I generated the page and it works like this:
```suggestion
    * Instead, peer reviewed lists of packages, where maintainers can take their `quality declaration <#reference-implementation>`_ and have their peers verify that their claims/justifications are truthful/reasonable, respectively, can be created 
```

> The outcome of this REP should be that maintainers have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications.
> The goal being to improve documentation of these policies and improve communication of these quality characteristics between maintainers to consumers.

Reviewer: ```suggestion
```

This REP describes a set of categories meant to convey the quality or maturity of packages in the ROS ecosystem.
Inclusion in a category, or quality level, is based on the policies to which a package adheres.
The categories address policies about versioning, change control, documentation, testing, dependencies, platform support and security.

Motivation
==========

The purpose of this REP is to provide standard guidelines for evaluating package quality, for both maintainers and consumers.

Package maintainers can use these guidelines to establish and document their own specific policies addressing how their packages achieve certain quality levels.

Consumers can use the guidelines in the REP, as well as the corresponding policies defined by maintainers, to set expectations on package quality and better evaluate whether a package or its dependencies meet the standards of use in their projects.

> The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their practices address each, adjusted their policies if needed, and then documented their policies and justifications. 

Reviewer 2: I'm not clear on the definitions of "policy" and "practice" in this sentence. Does this work better?

```suggestion
The outcome of this REP should be that maintainers who want to evaluate their packages' quality have looked at each guideline, thought about how their policies address each, adjusted their policies if needed, and then documented their policies and justifications. 
```

This REP describes a set of categories meant to convey the quality or maturity of packages in the ROS ecosystem.
Inclusion in a category, or quality level, is based on the policies to which a package adheres.

> The categories address policies about things including version, change control, documentation, testing, dependencies, platform support and others.

Reviewer: ```suggestion
The categories address policies about versioning, change control, documentation, testing, dependencies, platform support and security.
```
